src/cryptonote_core/cryptonote_core.cpp

```diff
@@ -180,6 +180,14 @@ namespace cryptonote
     "service-node"
   , "Run as a service node"
   };
+  static const command_line::arg_descriptor<std::string> arg_public_ip = {
+    "public-ip"
+  , "Globally reachable IP address for incoming storage server requests (required for Service Nodes)"
```

Comment:
This will be used for more than just storage server requests in the future -- lokinet and blink quorum communication will also make use of it.

Reply:
Updated.